lib/minify.js

```diff
@@ -0,0 +1,168 @@
+        try {
+            var cache = exports.readFile(filename);
+            cache = JSON.parse(cache)[key];
+            if (!cache) throw "init";
+            cache.props = Dictionary.fromObject(cache.props);
+        } catch(ex) {
+            cache = {
+                cname: -1,
+                props: new Dictionary()
+            };
+        }
+    }
+    return cache;
+};
+
+var writeNameCache = function(filename, key, cache) {
+    if (filename) {
+        var data;
+        try {
+            data = exports.readFile(filename);
+            data = JSON.parse(data);
+        } catch(ex) {
+            data = {};
+        }
+        data[key] = {
+            cname: cache.cname,
+            props: cache.props.toObject()
+        };
+        exports.writeFile(filename, JSON.stringify(data, null, 2));
+    }
+};
+
+var minify = function(files, options) {
+    options = defaults(options, {
+        spidermonkey     : false,
+        outSourceMap     : null,
+        sourceRoot       : null,
+        inSourceMap      : null,
+        sourceMapUrl     : null,
+        sourceMapInline  : false,
+        fromString       : false,
+        warnings         : false,
+        mangle           : {},
+        mangleProperties : false,
+        nameCache        : null,
+        output           : null,
+        compress         : {},
+        parse            : {}
+    });
+    base54.reset();
+
+    // 1. parse
+    var toplevel = null,
+        sourcesContent = {};
+
+    if (options.spidermonkey) {
+        toplevel = AST_Node.from_mozilla_ast(files);
+    } else {
+        var addFile = function(file, fileUrl) {
+            var code = options.fromString
+                ? file
+                : exports.readFile(file);
+            sourcesContent[fileUrl] = code;
+            toplevel = parse(code, {
+                filename: fileUrl,
+                toplevel: toplevel,
+                bare_returns: options.parse ? options.parse.bare_returns : undefined
+            });
+        }
+        if (!options.fromString) files = exports.simple_glob(files);
+        [].concat(files).forEach(function (files, i) {
+            if (typeof files === 'string') {
+                addFile(files, options.fromString ? i : files);
+            } else {
+                for (var fileUrl in files) {
+                    addFile(files[fileUrl], fileUrl);
+                }
+            }
+        });
+    }
+    if (options.wrap) {
+      toplevel = toplevel.wrap_commonjs(options.wrap, options.exportAll);
+    }
+
+    // 2. compress
+    if (options.compress) {
+        var compress = { warnings: options.warnings };
+        merge(compress, options.compress);
+        toplevel.figure_out_scope();
+        var sq = Compressor(compress);
+        toplevel = sq.compress(toplevel);
+    }
+
+    // 3. mangle properties
+    if (options.mangleProperties || options.nameCache) {
+        options.mangleProperties.cache = readNameCache(options.nameCache, "props");
```

Comment:
`options.mangleProperties` defaults to false, which isn't an object :/ Need to know how this stuff is working before I can fix this...